crates/relayer-framework/src/tests/relayer_mock/contexts/chain.rs

```diff
@@ -0,0 +1,118 @@
+use alloc::string::{String, ToString};
+use std::sync::Mutex;
+use std::{collections::HashMap, sync::Arc};
+use tokio::runtime::Runtime;
+
+use ibc_relayer_runtime::tokio::context::TokioRuntimeContext;
+
+use crate::base::one_for_all::traits::chain::OfaChainTypes;
+use crate::base::one_for_all::traits::runtime::OfaRuntimeContext;
+use crate::tests;
+use crate::tests::relayer_mock::base::error::Error;
+use crate::tests::relayer_mock::base::types::{height::Height, packet::PacketKey, state::State};
+
+pub type ChainState = Arc<Mutex<HashMap<Height, State>>>;
+pub type ClientId =
+    <tests::relayer_mock::contexts::chain::MockChainContext as OfaChainTypes>::ClientId;
+
+/// The 'state' represents the received messages and received acks at a given height.
+/// The 'consensus_states' represent the known whole states of clients.
+/// Usually the 'consensus_states' would be the Merkle Tree root hashes, but for
+/// the MockChain the whole state is used to avoid needing to implement Merkle
+/// Trees and Proofs.
+pub struct MockChainContext {
+    pub name: String,
+    pub states: Arc<Mutex<HashMap<Height, State>>>,
+    pub consensus_states: Arc<Mutex<HashMap<ClientId, ChainState>>>,
```

Comment:
The `consensus_states` should be a `HashMap<ClientId, HashMap<Height, ChainState>>`, since a client can have many consensus states at different heights.

Let's also try and explain why we want what the `states` and `consensus_states` are for, and why they have heights and client IDs as the hash map keys.